service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java

```diff
@@ -141,11 +201,22 @@ private void initializeMessageHandlers() throws Exception {
             exec.execute(new BamCompleteHandler(bamCompletePhaser));
         }
         bamCompletePhaser.arriveAndAwaitAdvance();
+
+        // qc complete handler
+        qcCompleteHandlerShutdownLatch = new CountDownLatch(NUM_TEMPO_MSG_HANDLERS);
+        final Phaser qcCompletePhaser = new Phaser();
+        qcCompletePhaser.register();
+        for (int lc = 0; lc < NUM_TEMPO_MSG_HANDLERS; lc++) {
+            qcCompletePhaser.register();
+            exec.execute(new QcCompleteHandler(qcCompletePhaser));
+        }
+        qcCompletePhaser.arriveAndAwaitAdvance();
```

Comment:
Could we discuss these concepts during our next 1:1? (e.g. threads, latch, phaser)